### PR TITLE
Fix unit test build

### PR DESCRIPTION
**Pull request by cjolivier01**
## Description ##

SInce some other version of this fix (which I had in my personal branch) was requiring mshadow change (I dont know why, but per larroy email to dev@)

### Essentials ###
- [ ] Passed code style checking (`make lint`)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: tests/cpp/misc/memory_test.cc [hunk 1]
```diff
@@ -64,40 +64,40 @@ TEST(MEMORY_TEST, MemsetAndMemcopyPerformance) {
     std::cout << "====================================" << std::endl
               << "Data size: " << test::pretty_num(test_size) << std::endl << std::flush;
 
-    std::unique_ptr<uint8_t> buffer_1(new uint8_t[test_size]), buffer_2(new uint8_t[test_size]);
-    uint8_t *src = buffer_1.get(), *dest = buffer_2.get();
+    std::unique_ptr<float> buffer_1(new float[test_size]), buffer_2(new float[test_size]);
```
Comment (nswamy) [inline tests/cpp/misc/memory_test.cc]: why change from unit_8 to float? same question in other places.

Comment (cjolivier01) [reply tests/cpp/misc/memory_test.cc]: so that memcpy (for example) similates the more common use-case of float items rather than byte count

Comment (larroy) [reply tests/cpp/misc/memory_test.cc]: It's a bug to use new[] on a unique_ptr. This is undefined behaviour.

Comment (lebeg) [reply tests/cpp/misc/memory_test.cc]: The template parameter should be float[], not just float

**Comment by lebeg** [comment]
Actually it's fine to use [] on a unique_ptr: http://www.cplusplus.com/reference/memory/unique_ptr/operator[]/

**Comment by larroy** [comment]
I know, but you have to put it explicitly on the template. I checked that there's a template specialization for arrays.

**Pull request merged**